lib/elm-test.js

```diff
@@ -1,330 +1,163 @@
-  } else {
-    const root = process.cwd();
-
-    globs = globifyWithRoot(root, 'test?(s)/**/*.elm');
-  }
-
-  return flatMap(globs, resolveFilePath);
-}
+// Recursively search directories for *.elm files, excluding elm-stuff/
+function resolveFilePath(elmFilePathOrDir /*: string */) /*: Array<string> */ {
+  const candidates = !fs.existsSync(elmFilePathOrDir)
+    ? []
+    : fs.lstatSync(elmFilePathOrDir).isDirectory()
+    ? flatMap(
+        glob.sync('/**/*.elm', {
+          root: elmFilePathOrDir,
+          nocase: true,
+          ignore: '/**/elm-stuff/**',
+          nodir: true,
+        }),
+        resolveFilePath
+      )
+    : [path.resolve(elmFilePathOrDir)];
 
-function getGlobsToWatch(elmJson) {
-  let sourceDirectories;
-  if (elmJson['type'] === 'package') {
-    sourceDirectories = ['src'];
-  } else {
-    sourceDirectories = elmJson['source-directories'];
-  }
-  return [...sourceDirectories, 'tests'].map(function (sourceDirectory) {
-    return path.posix.join(sourceDirectory, '**', '*.elm');
-  });
+  // Exclude everything having anything to do with elm-stuff
+  return candidates.filter(
+    (candidate) => !candidate.split(path.sep).includes('elm-stuff')
+  );
 }
 
-let report;
-
-if (
-  args.report === 'console' ||
-  args.report === 'json' ||
-  args.report === 'junit'
-) {
-  report = args.report;
-} else if (args.report !== undefined) {
-  console.error(
-    "The --report option must be given either 'console', 'junit', or 'json'"
+function getGlobsToWatch(elmJson /*: any */) /*: Array<string> */ {
+  const sourceDirectories =
+    elmJson.type === 'package' ? ['src'] : elmJson['source-directories'];
+  return [...sourceDirectories, 'tests'].map((sourceDirectory) =>
+    path.posix.join(sourceDirectory, '**', '*.elm')
   );
-  process.exit(1);
-} else {
-  report = 'console';
 }
 
-function infoLog(msg) {
+function infoLog(
+  report /*: typeof Report.Report */,
+  msg /*: string */
+) /*: void */ {
   if (report === 'console') {
     console.log(msg);
   }
 }
 
-// It's important to globify all the arguments.
-// On Bash 4.x (or zsh), if you give it a glob as its last argument, Bash
-// translates that into a list of file paths. On bash 3.x it's just a string.
-// Ergo, globify all the arguments we receive.
-const isMake = args._[0] === 'make';
-const testFileGlobs = isMake ? args._.slice(1) : args._;
-const testFilePaths = resolveGlobs(testFileGlobs);
-const projectRootDir = process.cwd();
-const generatedCodeDir = Compile.getGeneratedCodeDir(projectRootDir);
-const hasBeenGivenCustomGlobs = testFileGlobs.length > 0;
-
-const elmJsonPath = path.resolve(path.join(projectRootDir, 'elm.json'));
-let projectElmJson;
-
-try {
-  projectElmJson = fs.readJsonSync(elmJsonPath);
-} catch (err) {
-  console.error('Error reading elm.json: ' + err.message);
-  throw process.exit(1);
+function clearConsole() {
+  process.stdout.write(
+    process.platform === 'win32' ? '\x1B[2J\x1B[0f' : '\x1B[2J\x1B[3J\x1B[H'
+  );
```

Comment:
What are the advantages of this over https://nodejs.org/api/console.html#console_console_clear?

Reply:
I didn’t write this code, but as far as I can tell `console.clear()` makes it _look_ like the screen was cleared, but if you scroll up all the previous printed stuff is there, while this one clears the screen _and kills the scrollback._

Try it!

```
node -e 'console.clear()'
```

vs

```
node -e "process.stdout.write('\x1B[2J\x1B[3J\x1B[H')"
```

Reply:
aha yes. I did try it and I see what you mean. Thanks!